system/modules/news/modules/ModuleNewsList.php

```diff
@@ -162,4 +162,60 @@ protected function compile()
 
 		$this->Template->archives = $this->news_archives;
 	}
+
+	/**
+	 * Counts the total matching items
+	 *
+	 * @param   array $newsArchives
+	 * @param   boolean $blnFeatured
+	 *
+	 * @return  int
+	 */
+	protected function countItems($newsArchives, $blnFeatured)
+	{
+		// HOOK: add custom logic
+		if (isset($GLOBALS['TL_HOOKS']['newsListCountItems']) && is_array($GLOBALS['TL_HOOKS']['newsListCountItems']))
+		{
+			foreach ($GLOBALS['TL_HOOKS']['newsListCountItems'] as $callback)
+			{
+				$intResult = \System::importStatic($callback[0])->$callback[1]($newsArchives, $blnFeatured, $this);
+
+				if ($intResult !== false && is_int($intResult))
+				{
+					return $intResult;
+				}
+			}
+		}
+
+		return \NewsModel::countPublishedByPids($newsArchives, $blnFeatured);
+	}
+
+	/**
+	 * Fetches the matching items
+	 *
+	 * @param   array $newsArchives
+	 * @param   boolean $blnFeatured
+	 * @param   int $limit
+	 * @param   int $offset
+	 *
+	 * @return  \Model\Collection|\NewsModel|null
+	 */
+	protected function fetchItems($newsArchives, $blnFeatured, $limit, $offset)
+	{
+		// HOOK: add custom logic
+		if (isset($GLOBALS['TL_HOOKS']['newsListFetchItems']) && is_array($GLOBALS['TL_HOOKS']['newsListFetchItems']))
+		{
+			foreach ($GLOBALS['TL_HOOKS']['newsListFetchItems'] as $callback)
+			{
+				$objCollection = \System::importStatic($callback[0])->$callback[1]($newsArchives, $blnFeatured, $limit, $offset, $this);
+
+				if ($objCollection !== false && ($objCollection === null || $objCollection instanceof \Model\Collection))
```

Comment:
Because `null` is a valid return type. There are 3 use cases:

1. I am responsible for this module and I return a collection
2. I am responsible for this module and I return `null` because nothing was found
3. I am not responsible

`false` is for case 3.